### ISSUE TITLE
`ipfs ls` should behave like `ls` when called on individual files.

### ISSUE THREAD

**Issue by Stebalien** [ipfs/go-ipfs#4186]
#### Version information:

> go-ipfs version: 0.4.11-dev-48476b292
Repo version: 5
System version: amd64/linux
Golang version: go1.8.3

#### Type:

Bug?

#### Severity:

Very Low

#### Description:

In the past, `ipfs ls /ipfs/Qm.../file.txt` listed the internal blocks in the file. Now, it just prints an error. In the near future, it looks like it will go back to listing internal blocks (restoring the original behavior).

Really, it should behave like the standard unix `ls` command and print the filename (if available in this context) and, in the future when we support them, any unix permissions on the file. Basically, internally, we should be listing the parent directory and then filtering out all files except the target file.

The tricky part is what to do when we *don't* know the parent directory (when the file is named by hash). I'd be inclined to print the file hash as the file name and set the permissions to 0777.

**Comment by Kubuxu** [comment]
The permission might be better being `444`.

**Comment by Stebalien** [comment]
Ah. Good point but I'd say 555 (+x) by default for convenience. Otherwise, one wouldn't be able to, e.g., execute `/ipfs/QmId...`.

**Comment by Kubuxu** [comment]
You are not able to execute it from fuse interface right now either way.

**Comment by Stebalien** [comment]
True. But once start storing an executable bit, I'd mark bare files as executable.
